uniswap-v2/src/modules/7_store_usd_prices.rs

```diff
@@ -0,0 +1,170 @@
+                    let mut token1_price = BigDecimal::zero();
+
+                    let token0_price =
+                        get_price_from_native_store(ordinal, &token0.address, &native_prices_store);
+
+                    if reserve0
+                        .mul(token0_price.clone())
+                        .ge(&min_liquidity_threshold)
+                    {
+                        token1_price = token1_derived_price * token0_price
+                    }
+
+                    output_store.set(
+                        ordinal,
+                        StoreKey::TokenPrice.get_unique_pool_key(&token1.address),
+                        &token1_price,
+                    );
+                }
+            }
+            _ => {}
+        }
+    }
+}
+
+fn get_price_from_native_store(
+    ordinal: u64,
+    token_address: &String,
+    native_prices_store: &StoreGetBigDecimal,
+) -> BigDecimal {
+    if token_address.eq(constants::WETH_ADDRESS) {
+        return get_weth_price_in_usd(ordinal, native_prices_store);
+    }
+    if constants::STABLE_COINS.contains(&token_address.as_str()) {
+        return BigDecimal::one();
+    }
+
+    for address in constants::PAIR_COINS.into_iter() {
+        let whitelisted_token = address.to_string();
+
+        let mut token_price = native_prices_store
+            .get_at(
+                ordinal,
+                StoreKey::TokenPrice.get_unique_pair_key(token_address, &whitelisted_token),
+            )
+            .unwrap_or(BigDecimal::zero());
+
+        if token_price.le(&BigDecimal::zero()) {
+            continue;
+        }
+
+        if whitelisted_token.eq(constants::WETH_ADDRESS) {
+            token_price = token_price * get_weth_price_in_usd(ordinal, native_prices_store)
+        }
+
+        if min_pool_liquidity_check(
+            ordinal,
+            token_address,
+            &whitelisted_token,
+            native_prices_store,
+        ) {
+            return token_price;
+        }
+    }
+
+    return BigDecimal::zero();
+}
+
+fn min_pool_liquidity_check(
+    ordinal: u64,
+    address1: &String,
+    address2: &String,
+    native_prices_store: &StoreGetBigDecimal,
+) -> bool {
+    let mut amount_locked_usd = native_prices_store
+        .get_at(
+            ordinal,
+            StoreKey::TokenBalance.get_unique_pair_key(address1, address2),
+        )
+        .unwrap_or(BigDecimal::zero());
+
+    if address2.eq(constants::WETH_ADDRESS) {
+        amount_locked_usd = amount_locked_usd * get_weth_price_in_usd(ordinal, native_prices_store);
+    }
+
+    if amount_locked_usd.ge(&BigDecimal::from_str("5000").unwrap()) {
+        return true;
+    }
+
+    return false;
+}
+
+fn get_weth_price_in_usd(ordinal: u64, native_prices_store: &StoreGetBigDecimal) -> BigDecimal {
+    native_prices_store
+        .get_at(
+            ordinal,
+            StoreKey::TokenPrice.get_unique_pair_key(
+                &constants::WETH_ADDRESS.to_string(),
```

Comment:
Use `constants::WETH_ADDRESS` instead of `&constants::WETH_ADDRESS.to_string()`

Reply:
Can you change the signature for it? (All other calls will be unaffected by the change as there is auto coersion from &String to &str

Reply:
Ah, I see, that works. Thanks